src/crimson/common/fatal_signal.cc

```diff
@@ -39,7 +39,8 @@ void FatalSignal::install_oneshot_signal_handler()
     if (static std::atomic_bool handled{false}; handled.exchange(true)) {
       return;
     }
-    FatalSignal::signaled(sig, info);
+    assert(info);
```

Comment:
we should therefore guarantee that we never reach here with a signal that does not have 'SA_SIGINFO' set.
Do we want to commit to that?

Reply:
I believe it's already guaranteed in `master`. This handler is used solely for the signals installed by the `::sigaction` call at line `48` and the `sa_flags` there always contains `SA_SIGINFO`.